File: frontend/src/features/inventory/inventory-queries.ts
export const inventoryBalancesKey = (
  userId: string,
  workspaceId: string,
  warehouseId = "",
  productId = ""
) =>
  [
    "inventory",
    "balances",
    userId,
    workspaceId,
    warehouseId,
    productId,
  ] as const

export const inventoryBalancesKeyPrefix = (
  userId: string,
  workspaceId: string
) => ["inventory", "balances", userId, workspaceId] as const

File: backend/src/InventoryFlow.Application/Features/Inventory/InventoryQueries.cs
using MediatR;

namespace InventoryFlow.Application.Features.Inventory;

/// <summary>Lists current workspace inventory balances.</summary>
public sealed record ListInventoryBalancesQuery(Guid WorkspaceId, Guid? WarehouseId, Guid? ProductId)
    : IRequest<IReadOnlyList<InventoryBalanceResponse>>;
